daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -328,10 +337,11 @@ class VariableMap private(vTable: Map[GlobalQName, ArrayBuffer[VariableInstance]
         // Evaluate defineVariable statements with non-constant default value expressions
         case (VariableDefined, DataValue.NoValue, true) => {
           val res = inst.defaultValueExpr.get.evaluate(state)
-          inst.setValue(DataValue.unsafeFromAnyRef(res))
+          inst.setDefaultValue(DataValue.unsafeFromAnyRef(res))
         }
         case (_, _, _) => // Do nothing
       }
+      inst.firstInstanceInitialValue = inst.value
```

Comment:
Personally, I'd rather see a function called after ``forceExprssionEvaluations`` is called that does this copy. That way there is only a single place where this variable is changed and it's easier to verify it's happening at the right time. Functionally it's exactly the same, but I think eaiser to read and understand what's going on.